pkg/registry/apis/secret/reststorage/keeper_rest.go

```diff
@@ -212,9 +212,26 @@ func ValidateKeeper(keeper *secretv0alpha1.Keeper, operation admission.Operation
 		return errs
 	}
 
-	// TODO: validation of SQL keeper, once we have a finalized spec.
-	if keeper.Spec.SQL != nil {
-		_ = keeper.Spec.SQL
+	// TODO: Improve SQL keeper validation.
+	// SQL keeper is not allowed to use `secureValueName` in credentials fields to avoid depending on another keeper.
+	if keeper.Spec.SQL != nil && keeper.Spec.SQL.Encryption != nil {
```

Comment:
Maybe a method that returns bool (e.g. `IsSqlKeeper() bool`) would make this easier to understand

Reply:
Done! If there's something else let me know and I will address in a follow up PR 🚀